test/util/geo.test.cpp

```diff
@@ -6,6 +6,11 @@
 
 using namespace mbgl;
 
+TEST(LatLng, MixedCoordinateTypes) {
+    static_assert(std::is_assignable<LatLng, ScreenCoordinate>::value == false, "LatLng and ScreenCoordinate cannot be interchangeable.");
+    static_assert(std::is_assignable<LatLng, ProjectedMeters>::value == false, "LatLng and ProjectedMeters cannot be interchangeable.");
+}
```

Comment:
Try it the other way around -- assigning a `LatLng` to a `ScreenCoordinate` variable, or passing a `LatLng` to a function that expects `ScreenCoordinate`.